cmd/kubelet/app/server.go

```diff
@@ -597,6 +622,94 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 	return nil
 }
 
+func updateTransport(clientConfig *restclient.Config, clientCertificateManager certificate.Manager) error {
+	pool := x509.NewCertPool()
```

Comment:
you shouldn't replicate building the CA pool... `TLSConfigFor` already does that for you